### ISSUE TITLE
Watch mode doesn't watch all files

### ISSUE THREAD

**Issue by joenye** [marp-team/marp-cli#289]
It seems only changes to `*.md` files will be watched. For example, changes to images will not be watched.

### Steps to reproduce:

1. Create a simple `index.md` that references `image.png` on a slide.
2. Execute `marp -psw .`
3. Change `index.md` -> Preview updates :ballot_box_with_check:
4. Change `image.png` -> Preview does not update :negative_squared_cross_mark:

**Comment by yhatt** [comment]
Currently Marp CLI is observing the change only for Markdown file and theme CSS file because watching image resources using in Markdown requires extra parse step.

I think it's hard to implement, but also must have a worth.

**Comment by joenye** [comment]
I recommend looking at [chokidar](https://github.com/paulmillr/chokidar). You can reliably watch the user-provided directory and handle changes to _all_ files accordingly.

**Comment by yhatt** [comment]
Yes. Probably we will take the simplest one at the first: should observe all files in specified directory by `--input-dir` / `-I` option.

In server mode, we are internally being inherited the passed directory to `--input-dir` option, so observer still will work in the server directory.

The fact we should consider is that the current watch mode can work in both of regular file conversion and server mode. When observer is working in regular conversion mode and the converted file was created into observed directory, we must avoid infinite loop.

**Comment by simoneggler** [comment]
this would be a really helpful feature

**Comment by mrtndwrd** [comment]
I noticed `--watch` also doesn't watch a custom engine passed by `--engine`. Maybe that could be a simple one to add, just like how the theme is watched. I'm currently rerunning watch/build every time I change my engine (which contains a bunch of custom logic to add data to mermaid charts).
